lib/schema_registry/client.rb

```diff
@@ -48,28 +50,69 @@ def subject(name)
     end
 
     def default_compatibility_level
-      request(:get, "/config")["compatibilityLevel"]
+      request(:get, '/config')['compatibilityLevel']
     end
 
     def default_compatibility_level=(level)
-      request(:put, "/config", compatibility: level)
+      request(:put, '/config', compatibility: level)
+    end
+
+    # Build options hash for net/http based on params provided. Primary for selectivly adding TLS config options for MTLS
+    def self.connection_options(**config)
+      options = {}
+
+      unless config[:verify_mode].nil?
+        options[:verify_mode] = OpenSSL::SSL.const_get(config[:verify_mode].upcase)
+      end
+
+      unless config[:ca_certificate].nil?
+        if File.exist?(config[:ca_certificate])
+          options[:ca_file] = config[:ca_certificate]
+        else
+          raise ArgumentError, "ca file not found [#{config[:ca_certificate]}]"
+        end
+      end
+
+      unless config[:client_key].nil?
+        if File.exist?(config[:client_key])
+          options[:key] = OpenSSL::PKey::RSA.new(File.read(config[:client_key]))
+        else
+          raise ArgumentError, "client key file not found [#{config[:client_key]}]"
+        end
+      end
+
+      unless config[:client_certificate].nil?
+        if File.exist?(config[:client_certificate])
+          options[:cert] = OpenSSL::X509::Certificate.new(File.read(config[:client_certificate]))
+        else
+          raise ArgumentError, "client cert file not found [#{config[:client_certificate]}]"
+        end
+      end
+
+      options
     end
 
     def request(method, path, body = nil)
-      Net::HTTP.start(endpoint.host, endpoint.port, use_ssl: endpoint.scheme == 'https') do |http|
+      # build config for http client
+      default_options = {
+        use_ssl: endpoint.scheme == 'https',
+        verify_mode: @verify_mode
```

Comment:
I think this is a leftover from a previous iteration.

Reply:
🤦‍♂️

Reply:
pushed

Reply:
@wvanbergen  thanks, just pushed a fix